components/SwitchNetworkModal/SwitchNetworkModal.tsx

```diff
@@ -50,14 +51,15 @@ const SwitchNetworkModal: FC<SwitchNetworkModalProps> = ({ pool, poolNetwork, is
 		);
 
 		try {
+			if (!walletAddress) {
+				setSendModalBack(true);
```

Comment:
The switch modal has to be sent back, otherwise it will be on top of the "connectWallet" modal